e2e:serial: check scheduling error per topology scope

Tal-or opened the pull request:
When the scheduler plugin failed to align a pod into a node it reports an error.
The error message is varies upon the `TopologyManagerScope` configuration.

We should support a check of the scheduling error for both container and pod scope configurations.
This patch adds the support for the container scope (pod scope support is already exists)

Signed-off-by: Talor Itzhak <titzhak@redhat.com>

fromanirh commented on test/e2e/serial/tests/workload_unschedulable.go:
```diff
@@ -239,7 +242,9 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload unsched
 			}
-			nodeSelector := map[string]string{}
+			nodeSelector := map[string]string{
+				serialconfig.MultiNUMALabel: "2",
```
do we need this change in this patch?

Tal-or replied:
Yes. In case there are none multi numa machines in the cluster (usually developer use-case) the daemonset will place a pod in there. 

Usually, the single NUMA machines are not strong enough and they got filtered by the `NodeResourceFit` plugin which in turn leads to test failure because we're looking for the `NodeTopologyMatch` error message and are unable to find it.

fromanirh replied:
Uhm, makes sense but please split this change in its own commit (please copy/paste [a possibly enhanced version of] this explanation there)

Tal-or replied:
Done.

fromanirh commented on test/e2e/serial/tests/workload_unschedulable.go:
```diff
@@ -334,7 +339,9 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload unsched
 			}
-			nodeSelector := map[string]string{}
+			nodeSelector := map[string]string{
+				serialconfig.MultiNUMALabel: "2",
```
ditto

Tal-or commented:
Tested locally (only the unschedulable tests).
The tests were passed and not skipped.

fromanirh commented on test/utils/nrosched/nrosched.go:
```diff
@@ -117,8 +119,12 @@ func CheckNROSchedulerAvailable(cli client.Client, NUMAResourcesSchedObjName str
 
-func IsSchedulingErrorCannotAlign(msg string) bool {
-	return strings.Contains(msg, ErrorCannotAlignPod)
+func IsSchedulingErrorCannotAlign(msg, policy string) bool {
```
let's inline this code in `IsSchedulingErrorCannotAlign` - in hindsight, this helper should never have been extracted (let alone made public) in the first place

Tal-or replied:
Done.

ffromani reviewed (commented):
Nice, and we're almost there!
mostly questions and possible improvement inside

Tal-or committed:
e2e:serial: check scheduling error per topology scope

When the scheduler plugin failed to align a pod into a node it reports an error.
The error message is varies upon the  configuration.

We should support a check of the scheduling error for both container and pod scope configurations.
This patch adds the support for the container scope (pod scope support is already exists)

Signed-off-by: Talor Itzhak <titzhak@redhat.com>

Tal-or committed:
e2e:serial: place pods only on multi NUMA nodes

There are times when single numa machines are in the cluster (usually developer use-case).
By default, if the machine is a worker, the daemonset will try to place a pod in there.

Usually, single NUMA machines have fewer resources and they are getting filtered by the `NodeResourceFit` plugin, hence they'll never reach the `NodeTopologyMatch` filtering process, which in turn leads to test failure.

This is because the test expects the pod to be rejected by the `NodeTopologyMatch` plugin and it's looking for the plugin's error message, but unable to find it.

Signed-off-by: Talor Itzhak <titzhak@redhat.com>

Tal-or commented:
> this seems legit. I'll check later.

The error or the code?

fromanirh commented:
so, the error is indeed legit and becomes evident if we rebase on top of the `main` branch. There are multiple way to fix, I don't have strong feelings. Shamelessly went for the easiest:
```
$ git diff
diff --git a/test/e2e/serial/tests/workload_unschedulable.go b/test/e2e/serial/tests/workload_unschedulable.go
index e489b919..39486671 100644
--- a/test/e2e/serial/tests/workload_unschedulable.go
+++ b/test/e2e/serial/tests/workload_unschedulable.go
@@ -384,7 +384,7 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload unsched
                                        Expect(err).ToNot(HaveOccurred(), "unable to get pod %s/%s to be Running after %v", pod.Namespace, pod.Name, podRunningTimeout)
 
                                } else {
-                                       isFailed, err := nrosched.CheckPODSchedulingFailedForAlignment(fxt.K8sClient, pod.Namespace, pod.Name, serialconfig.Config.SchedulerName, tmPolicy)
+                                       isFailed, err := nrosched.CheckPODSchedulingFailedForAlignment(fxt.K8sClient, pod.Namespace, pod.Name, serialconfig.Config.SchedulerName, string(tmPolicy))
                                        if err != nil {
                                                _ = objects.LogEventsForPod(fxt.K8sClient, pod.Namespace, pod.Name)
                                        }

```

ffromani reviewed (commented):
/lgtm
/approve

fromanirh commented:
/cherry-pick release-4.11
